rmw_iceoryx_cpp/src/rmw_init.cpp

```diff
@@ -95,8 +96,9 @@ rmw_init(const rmw_init_options_t * options, rmw_context_t * context)
   context->impl = nullptr;
 
   // create a name for the process to register with the RouDi daemon
-  extern char * __progname;
-  auto progName = std::string(__progname);
+  rcutils_allocator_t allocator = rcutils_get_default_allocator();
+  auto progam_name = rcutils_get_executable_name(allocator);
+  auto progName = std::string(progam_name);
```

Comment:
I have no idea what this is doing. I guess for ROS specific changes it's better when someone does the reviews :)

Reply:
@elBoberido This change here is relatively trivial and does not require any ROS knowledge.

1. So far the name for the `PoshRuntime` was directly read from [the external UNIX-only variable](https://github.com/ros2/rmw_iceoryx/blob/humble/rmw_iceoryx_cpp/src/rmw_init.cpp#L97-L107)
2. The `rcutils` package which is already used widely in `rmw_iceoryx` offers a platform-independent method, which is now called. For UNIX-based systems nothing changes, the external UNIX-only variable is read. However, there is [also support for Windows](https://github.com/ros2/rcutils/blob/rolling/src/process.c#L56-L76).


Does that make it clear? Do you feel confident enough to review this MR or shall I search for someone else?

Reply:
Well, you basically told me what the names implied but that does not tell me whether this is the idiomatic approach or something which is basically deprecated because there is a `get_super_duper_foo` :)

Just a small side note on how I handle such PRs for the Rust bindings. For such small and obviously not incorrect changes I just create the PR to check if the CI is still green and then merge it. In such cases, the PRs are basically just there to track the changes for myself and not to accidentally break master. For larger changes I try to find someone but even then, when I'm confident and the CI is green I still merge them. But on the other hand, the Rust bindings are still in a premature state so this procedure might not be something you want to adopt.